server/src/main/java/org/elasticsearch/script/ScriptedMetricAggContexts.java

```diff
@@ -31,6 +33,25 @@
 import java.util.Map;
 
 public class ScriptedMetricAggContexts {
+    private static final DeprecationLogger DEPRECATION_LOGGER =
+        new DeprecationLogger(Loggers.getLogger(ScriptedMetricAggContexts.class));
+
+    // Public for access from tests
+    public static final String AGG_PARAM_DEPRECATION_WARNING =
+        "params._agg/_aggs for scripted metric aggregations are deprecated, use state/states (not in params) instead. " +
+        "Set system property es.aggregations.enableDeprecatedScriptedMetricAggParam = false to disable this deprecated behavior.";
+
+    public static boolean deprecatedAggParamEnabled() {
+        boolean enabled = Boolean.parseBoolean(
+            System.getProperty("es.aggregations.enableDeprecatedScriptedMetricAggParam", "true"));
```

Comment:
This should be named with underscores. I think the name doesn't need the "deprecated" in it.

Reply:
So `enable_scripted_metric_agg_param` or is that too generic?